src/components/checkbox/GroupableCheckboxReducersUtils.ts

```diff
@@ -0,0 +1,68 @@
+import * as _ from 'underscore';
+import {IReduxAction} from '../../utils/ReduxUtils';
+import {ICheckboxState} from './CheckboxReducers';
+import {IGroupableCheckboxActionPayload} from './GroupableCheckboxActions';
+import {groupableCheckboxInitialState, IGroupableCheckboxesState} from './GroupableCheckboxReducers';
+
+export const addParentCheckbox = (state: IGroupableCheckboxesState,
+                                  action: IReduxAction<IGroupableCheckboxActionPayload>): IGroupableCheckboxesState => {
+  const newCheckbox = {
+    id: action.payload.id,
+    checked: !!action.payload.checked,
+  };
+
+  if (state === groupableCheckboxInitialState) {
+    return {
+      ...state,
+      parentId: action.payload.id,
+      parent: newCheckbox,
+    };
+  }
+
+  state.parent = newCheckbox;
```

Comment:
You should not directly change the state. You have to send a copy of it. So here, you should return 
```
return {
  ...state,
  parent: newCheckbox,
};
```